### PR TITLE
Allow log.xxx.default to set logging settings for the default logger only

### PR DESCRIPTION
**Pull request by zeripath**
In #11269 a user wanted to set the default logger to WARN level but leave the rest of the loggers at Info. The current settings require that in order to set this the user would have to add:

```ini
[log.file.macaron]
LEVEL = Info

[log.file.router]
LEVEL = Info

[log.file.access]
LEVEL = Info
```

for each different type logger they want.

This PR allows the user to set `[log.file.default]` for settings for the default logger only

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: modules/setting/log.go [hunk 1]
```diff
@@ -261,9 +261,13 @@ func newLogService() {
 			continue
 		}
 
-		sec, err := Cfg.GetSection("log." + name)
+		fmt.Println(name)
```
Comment (guillep2k) [inline modules/setting/log.go]: Is this line intentional?

Comment (zeripath) [reply modules/setting/log.go]:
```suggestion
```

**Review by guillep2k — APPROVED** [review]

**Comment by silverwind** [comment]
Maybe add some info about this to `app.ini.sample` to make it more discoverable?

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Logging config still feels like black magic to me, but I guess this is fine.

**Comment by zeripath** [comment]
@silverwind it's really not difficult:

```ini
[log]
MODE = console
LEVEL = debug
ROOT_PATH = /home/andrew/src/go/gitea/log
REDIRECT_MACARON_LOG = true
MACARON = console
ROUTER = console
XORM = console
```

If, as is sensible, you don't want the SQL being logged - then set: 

```ini
[database]
LOG_SQL=false
```

(I agree it's silly that this isn't the default and that the logging of SQL should only occur at DEBUG level but that's a breaking change which we need to discuss and propose on a PR.)

Say you want to have a specific TRACE level logging for a particular file or area:
```ini
[log]
MODE = console, traceconsole
...

[log.traceconsole]
MODE=console
LEVEL=trace
EXPRESSION=compare
```

The expression will attempt a match against the filename:line:caller, or against the message itself

The difficulties come from if you want logging at level WARN or above only - in which case the additional comments I've put in to the document explain why.

```ini
[database]
LOG_SQL=false

...

[log]
MODE = console
LEVEL = info
ROOT_PATH = /home/andrew/src/go/gitea/log
REDIRECT_MACARON_LOG = true
MACARON = console
ROUTER = console
XORM = console

[log.console.default]
LEVEL=WARN
```

Then you will only get main gitea logs at WARN but still get ROUTER, MACARON, and XORM info logs.

**Pull request merged**